### PR TITLE
Add infra to e2e-runner and job configs to allow running tests with a skewed kubectl

### PR DESCRIPTION
**Pull request by ikehz**
Mostly reformatting & renaming to be less confusing; the meat is in `JENKINS_USE_SKEW_KUBECTL`.

### CODE REVIEW

**Comment by ihmccreery** [comment]
This would have been a lot easier if https://github.com/kubernetes/kubernetes/issues/8817 were fixed, but I didn't quite have the gusto to try to fix #8817 myself.

File: hack/jenkins/e2e-runner.sh [hunk 1]
```diff
@@ -307,20 +307,23 @@ if [[ "${E2E_UP,,}" == "true" ]]; then
...
-	# the rest of the e2es.
+        # Add a report prefix for the e2e tests so that the tests don't get overwritten when we run
+        # the rest of the e2es.
         E2E_REPORT_PREFIX='upgrade' e2e_test "${GINKGO_UPGRADE_TEST_ARGS:-}"
-	# If JENKINS_USE_OLD_TESTS is set, back out into the old tests now that we've upgraded.
-        if [[ "${JENKINS_USE_OLD_TESTS:-}" == "true" ]]; then
-            cd ../kubernetes_old
-        fi
+    fi
+    if [[ ! "${JENKINS_USE_SKEW_TESTS:-}" == "true" ]]; then
```
Comment (spxtr) [inline hack/jenkins/e2e-runner.sh]: `!=`

File: hack/jenkins/e2e-runner.sh [hunk 2]
```diff
@@ -307,20 +307,23 @@ if [[ "${E2E_UP,,}" == "true" ]]; then
 fi
 
 # Allow download & unpack of alternate version of tests, for cross-version & upgrade testing.
-if [[ -n "${JENKINS_PUBLISHED_TEST_VERSION:-}" ]]; then
+if [[ -n "${JENKINS_PUBLISHED_SKEW_VERSION:-}" ]]; then
```
Comment (spxtr) [inline hack/jenkins/e2e-runner.sh]: I'd appreciate a short comment on what these environment variables do.

**Comment by ihmccreery** [comment]
@spxtr PTAL.

**Comment by ihmccreery** [comment]
This should be manually merged with https://github.com/kubernetes/test-infra/pull/35.

**Comment by spxtr** [comment]
I appreciate the comments :)

LGTM

**Pull request merged**